### DISCUSSION TITLE
pod exec feature in argocd

### DISCUSSION THREAD

**Discussion by aman35** [argoproj/argo-cd#12411] [Q&A]
Hi, I have enabled the Pod terminal exec following https://argo-cd.readthedocs.io/en/stable/operator-manual/web_based_terminal/I have set exec.enabled key to "true" and added patched the argocd-server clusterRole to perform these actions.

We are using gitlab for authentication and gitlab Groups for authorization.

I have not added "p, role:myrole, exec, create, */*, allow" in argocd-rbac-cm. Infact there is no reference of exec resource in argocd-rbac-cm.. even though I am able to do pod exec and run commands in pod/containers from argocd UI.

I want this feature to enable for one group and disabled for other groups. Please help

**Comment by crenshaw-dev** [comment]
Can you please email reproduction steps and version information to [cncf-argo-security@lists.cncf.io](mailto:cncf-argo-security@lists.cncf.io)? If RBAC bypass is indeed possible, we'll need to address that as a security vulnerability.

**Reply by aman35** [reply]
@crenshaw-dev I just observed following line in argocd-rbac-cm is causing this behavior. 

  g, kubernetes, role:admin

when I change this line to   g, kubernetes, role:readonly then exec feature works as expected.  UI terminal only shows up for a msp group for which I have added following line in rbac-cm. For other group's users terminal tab does not even show up.

 p, role:msp, exec, create, */*, allow

but I am concerned what is significance of "g, kubernetes, role:admin", what would be effect of changing role to readonly from admin.
Can you please help me to understand the purpose of this line?

We are using single argocd cluster which manages multiple K8s Clusters.

**Reply by aman35** [reply]
The significance of "g, kubernetes, role:admin" is, it allows to sync application of applications. 
To overcome this, I changed it to readonly and give admin privileges to other group.
